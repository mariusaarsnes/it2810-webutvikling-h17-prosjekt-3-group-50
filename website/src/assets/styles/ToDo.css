.todo-list {
    background-color: white;
    width: 40%;
    margin-left: 10%;
    margin-top: 5%;

}

@font-face {
    font-family: IntroRust;
    src: url(../fonts/IntroRust.otf);
}

#yellow {
    color: goldenrod;
}

#titleToDos {
    font-family: IntroRust, sans-serif;
}

.liItem{
    list-style-type: none;
    font-family: IntroRust, sans-serif;
    font-size: 14px;
    height: 50px;
    margin: 20px;
    width: 100%;
    padding-top: 3%;
    -webkit-box-shadow: 0px 0px 51px -20px rgba(0,0,0,0.75);
    -moz-box-shadow: 0px 0px 51px -20px rgba(0,0,0,0.75);
    box-shadow: 0px 0px 51px -20px rgba(0,0,0,0.75);
}

.liItem:hover {
    background-color: #dddddd;
}


#todo-holder {
    width: 100%;
    padding-left: 0;
    margin-left: -20px;
    margin-top: 20px;
}

input[type=checkbox] {
    width: 1.8em;
    height: 1.8em;
    background-color: white;
    border-radius: 50%;
    vertical-align: middle;
    border: 1px solid #ddd;
    -webkit-appearance: none;
    outline: none;
    cursor: pointer;
    margin-right: 4%;
    margin-left: 3%;
}



input[type=checkbox]:hover {
    background-color:  goldenrod;
}

.toDo {
    display: flex;
}

.todo-info {
    font-size: 14px;

}

#btn-green {
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 10px 25px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 4px;
}

#btn-green:hover {
    background-color: darkgreen;
}

input[type=text] {
    width: 83%;
    padding: 13px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
}
