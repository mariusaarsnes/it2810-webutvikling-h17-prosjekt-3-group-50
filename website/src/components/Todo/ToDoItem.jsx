import React from 'react';

export class ToDoItem extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            checked: false
        };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        this.setState({checked: !this.state.checked});

    }

    render() {
        const text = this.state.checked;

        return (

            <li key={this.props.index} className="todo-item">
                <div className="category-todo" style={{backgroundColor: '' + this.props.color}}></div>
                <input type="checkbox" key={this.props.index} label={text} onClick={this.props.onClick}
                       onChange={() => this.handleClick()}/>
                <p className="todo-text">{this.props.value}</p>
            </li>
        );
    }
}
