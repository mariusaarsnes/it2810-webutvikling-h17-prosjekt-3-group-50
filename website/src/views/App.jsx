import React from 'react';
import {Switch, Route} from 'react-router-dom'
import '../assets/styles/App.css';
import {Navbar} from '../components';
import {Schedule,Appointment, Frontpage, NoteView, ToDoView} from '../views';


export class App extends React.Component {
    render() {
        return (
            <div className="body-fullwidth">
                <Navbar navbarLocation={"DASHBOARD"}/>
                <div id="content-container">
                    <Switch>
                        <Route exact path='/' component={Frontpage}/>
                        <Route exact path='/todo' component={ToDoView}/>
                        <Route exact path='/schedule' component={Schedule}/>
                        <Route exact path='/appointments' component={Appointment}/>
                        <Route exact path='/note' component={NoteView}/>
                        <Route exact path='/todo' component={ToDoView}/>
                    </Switch>
                </div>
            </div>
        );
    }
}
