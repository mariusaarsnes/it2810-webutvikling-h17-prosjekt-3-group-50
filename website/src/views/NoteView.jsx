import React from 'react';
import '../assets/styles/NoteView.css';
import {NoteContainer} from '../components/notes/NoteContainer.jsx'
import {CategoryContainer} from "../components/categories/CategoryContainer";
import {CategoryFilterContainer} from "../components/categories/CategoryFilterContainer";

export class NoteView extends React.Component {

    constructor(props) {
        super(props);
        //This method needs to be bound for it to filter
        this.filterNotes = this.filterNotes.bind(this);
        //This function is required in order to update the categoryFilter once we have added
        //A new category
        this.updateCategoryFilter = this.updateCategoryFilter.bind(this);
        this.state = {
            noteContainer: <NoteContainer ref={instance => {
                this.noteContainer = instance
            }} filterNotes={this.filterNotes}/>,

            categoryFilterContainer: <CategoryFilterContainer ref={instance => {
                this.categoryFilterContainer = instance
            }} filterNotes={this.filterNotes}/>,

            //Category saving is determined by the ID, so the ID needs to be unique
            categoryContainer: <CategoryContainer
                updateCategoryFilter={this.updateCategoryFilter} id={6}/>
        }
    }

    componentDidMount() {
        this.filterNotes();
    }

    render() {
        return (
            <div className="NoteView">
                <div className="Left-Page">
                    <p>This is your notes for now</p>
                    <div>
                        {this.state.categoryFilterContainer}
                    </div>
                    <div className="Middle-Page">
                        {this.state.noteContainer}
                    </div>
                    <div className="AddNote">
                        <button className="add-button" onClick={() => this.appendNote()}>Add note</button>
                    </div>
                </div>
                <div className="Right-Page">
                    {this.state.categoryContainer}
                </div>
            </div>
        );
    }

    appendNote() {
        if (this.categoryFilterContainer.state.selectedCategory !== undefined) {
            this.noteContainer.appendNote(this.categoryFilterContainer.state.selectedCategory.props.color);
        } else {
            alert("No category has been selected!");
        }
    }

    filterNotes() {
        const selectedCategory = this.categoryFilterContainer.state.selectedCategory;
        this.noteContainer.setState(prevState => {
            return {...prevState, displayedNotes: this.getFilteredNotes(selectedCategory)}
        });
    }

    getFilteredNotes(filter) {
        if (filter !== undefined)
            return this.noteContainer.state.notes.filter((note) => note.props.color === filter.props.color);
        return this.noteContainer.state.notes;
    }

    //Since our CategoryFilter is a 100% dumb component we load it from the CategoryContainer component
    updateCategoryFilter(colors) {
        this.categoryFilterContainer.addCategories(colors);
    }

}
