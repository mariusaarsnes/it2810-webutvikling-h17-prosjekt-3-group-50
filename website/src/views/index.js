export * from './frontpage.jsx';
export * from './App.jsx';
export * from './Appointment.jsx';
export * from './NoteView.jsx';
export * from './Schedule.jsx';
export * from './ToDoView.jsx';
export * from './Notification.jsx';
