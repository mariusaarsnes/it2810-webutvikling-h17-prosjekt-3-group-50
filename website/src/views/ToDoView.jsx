import React from 'react';
import '../assets/styles/ToDo.css';
import {ToDo} from '../components/Todo/ToDo.jsx';
import {Navbar} from '../components/Navbar.jsx';
import {Category} from '../components/Todo/Category.jsx';


export class ToDoView extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            current_color: null,
        };
        this.handleCategory = this.handleCategory.bind(this);
    }

    handleCategory(color){
        this.setState({current_color: color});
    }

    render() {
        return (
            <div>
                <Navbar/>
                <div id="display-todo">
                    <ToDo color={this.state.current_color}/>
                    <Category  handleCategory={(color) => this.handleCategory(color)}/>
                </div>
            </div>

        );
    }
}
