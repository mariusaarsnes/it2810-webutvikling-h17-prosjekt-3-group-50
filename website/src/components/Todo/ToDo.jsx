import React from 'react';
import {ToDoItem} from './ToDoItem.jsx';


export class ToDo extends React.Component {
    constructor(props) {
        super(props);
        let todos = [];
        let colors_todo = [];

        this.value = "";


        if (localStorage.getItem("ToDo") != null) {
            todos = JSON.parse(localStorage.getItem("ToDo"));
            colors_todo = JSON.parse(localStorage.getItem("Colors"));
        }
        this.state = {
            filter: '',
            current_color: '',
            color_data: colors_todo,
            data: todos,
            displayed_data: todos,
            displayed_colors: colors_todo,

        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleClicks = this.handleClicks.bind(this);
        this.filter = this.filter.bind(this);
    }

    filter() {
        let color = this.props.selectedColor();
        this.setState({filter: color});
        if(color === undefined) {
            color = "#016D91";
        }
        this.setState({current_color: color});
        let displayed_colors = this.props.selectedColor() === undefined ? this.state.color_data :
            this.state.color_data.filter((color, index) => color === this.props.selectedColor());
        let displayed_data = this.props.selectedColor() === undefined ? this.state.data :
            this.state.data.filter((todo, index) => this.state.color_data[index] === this.props.selectedColor());
        this.setState({displayed_colors: displayed_colors, displayed_data: displayed_data});
    }

    handleChange(event){
        this.value = event.target.value;
    }

    handleSubmit(event) {

        let todos = this.state.data;
        let colors = this.state.color_data;
        if (this.value.length > 0) {
            todos.push(this.value);
            colors.push(this.state.current_color);
                this.setState({ data: todos, color_data: colors},() => this.filter());

            if (typeof(Storage) !== "undefined" ) {
                localStorage.setItem("ToDo", JSON.stringify(todos));
                localStorage.setItem("Colors", JSON.stringify(colors));
            }
        }
        this.value = "";
        event.preventDefault();
    }

   handleClicks(index) {
        let todos = JSON.parse(localStorage.getItem("ToDo"));
        let colors = JSON.parse(localStorage.getItem("Colors"));
        let handledTodos = [];
        let handledColors = [];
        for (let i = 0; i < todos.length; i++) {
            if (i !== index) {
                handledTodos.push(todos[i]);
                handledColors.push(colors[i]);
            }
        }
       this.setState({ data: handledTodos, color_data: handledColors}, () => this.filter());
       localStorage.setItem("ToDo", JSON.stringify(handledTodos));
       localStorage.setItem("Colors", JSON.stringify(handledColors));
    }

    renderToDoItems() {
        return this.state.displayed_data.map((todo, index) =>
            <ToDoItem className="items" value={todo} key={index} color={this.state.displayed_colors[index]}
                      onClick={() => this.handleClicks(index)}/>
        );

    }

    render() {
        return (
            <div className="Body">
                <div className="todo-list">
                    <ul id="todo-holder">
                        {this.renderToDoItems()}
                    </ul>
                    <input id="todo-text-field" type="text" placeholder="Write your TODO"
                           onChange={this.handleChange}/>
                    <input type="submit" id="btn-green" value="Add" onClick={this.handleSubmit}/>
                </div>
            </div>
        );
    }
}


