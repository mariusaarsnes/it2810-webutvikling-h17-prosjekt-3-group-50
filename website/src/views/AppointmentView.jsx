import React from 'react';
import '../assets/styles/NoteView.css';
import {AppointmentContainer} from '../components/appointments/AppointmentContainer.jsx'
import {Categories} from "../components/categories/Categories";

export class AppointmentView extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            appointmentContainer: <AppointmentContainer ref={instance => {
                    this.appointmentContainer = instance
                }}/>
        }
    }

    render() {
        return (
            <Categories id={3} title={"appointments"} filter={() => this.appointmentContainer.filter()}>
                {this.state.appointmentContainer}
            </Categories>
        );
    }

}
