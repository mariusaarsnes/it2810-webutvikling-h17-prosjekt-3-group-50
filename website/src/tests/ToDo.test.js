//localStorage is not imported to be used as a class, but to be initiated with jest.
import localStorage from './localStorageMock';

import React from 'react';
import renderer from 'react-test-renderer';
import {ToDoView} from "../views/ToDoView";
import {ToDo} from "../components/Todo/ToDo";
import {ToDoItem} from "../components/Todo/ToDoItem";
import {CategoryItem} from "../components/Todo/Items";

/* Main function for testing the: */
/* todo application */
describe('ToDo Overview', () => {
    it('renders the whole Todo application', () => {
        const instance = renderer.create(<ToDoView />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

/* Test function for testing the: */
/* todo components */
describe('ToDo Components', () => {
    it('renders the Category Component', () => {
       const instance = renderer.create(<ToDo/>);
       const tree = instance.toJSON();
       expect(tree).toMatchSnapshot();
    });

    it('renders the CategoryItem Component', () => {
        const instance = renderer.create(<CategoryItem/>);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the ToDo Component', () => {
        const instance = renderer.create(<ToDo />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the ToDoItem Component', () => {
        const instance = renderer.create(<ToDoItem/>);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });
});