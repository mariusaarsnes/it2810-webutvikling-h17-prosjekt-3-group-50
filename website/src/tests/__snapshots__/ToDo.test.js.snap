// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo Components renders the Category Component 1`] = `
<div
  className="Body"
>
  <div
    className="todo-list"
  >
    <ul
      id="todo-holder"
    />
    <input
      id="todo-text-field"
      onChange={[Function]}
      placeholder="Write your TODO"
      type="text"
    />
    <input
      id="btn-green"
      onClick={[Function]}
      type="submit"
      value="Add"
    />
  </div>
</div>
`;

exports[`ToDo Components renders the CategoryItem Component 1`] = `
<li
  className="category-item"
>
  <input
    className="btn-category"
    onClick={undefined}
    style={
      Object {
        "backgroundColor": "undefined",
      }
    }
    type="submit"
    value=""
  />
</li>
`;

exports[`ToDo Components renders the ToDo Component 1`] = `
<div
  className="Body"
>
  <div
    className="todo-list"
  >
    <ul
      id="todo-holder"
    />
    <input
      id="todo-text-field"
      onChange={[Function]}
      placeholder="Write your TODO"
      type="text"
    />
    <input
      id="btn-green"
      onClick={[Function]}
      type="submit"
      value="Add"
    />
  </div>
</div>
`;

exports[`ToDo Components renders the ToDoItem Component 1`] = `
<li
  className="todo-item"
>
  <div
    className="category-todo"
    style={
      Object {
        "backgroundColor": "undefined",
      }
    }
  />
  <input
    className="todo-checkbox"
    label={false}
    onChange={[Function]}
    onClick={undefined}
    type="checkbox"
  />
  <p
    className="todo-text"
  />
</li>
`;

exports[`ToDo Overview renders the whole Todo application 1`] = `
<div
  className="NoteView"
>
  <div
    className="Left-Page"
  >
    <p>
      This is your TODO for now
    </p>
    <div>
      <div
        className="NoteHolder"
      >
        <p>
          <orange>
            Categories
          </orange>
           - Select to filter your notes and add new ones!
        </p>
        <div
          className="Notes"
        >
          <div
            className="CategoryFilter"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#016D91",
                "border": "none",
              }
            }
          />
          <div
            className="CategoryFilter"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#E53F6F",
                "border": "none",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="Middle-Page"
    >
      <div
        className="Body"
      >
        <div
          className="todo-list"
        >
          <ul
            id="todo-holder"
          />
          <input
            id="todo-text-field"
            onChange={[Function]}
            placeholder="Write your TODO"
            type="text"
          />
          <input
            id="btn-green"
            onClick={[Function]}
            type="submit"
            value="Add"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="Right-Page"
  >
    <div
      className="CategoryCreatorHolder"
    >
      <p>
        Categories
      </p>
      <div
        className="Category"
      >
        <div
          className="CategoryColor"
          style={
            Object {
              "backgroundColor": "#016D91",
            }
          }
        />
        <div>
          <input
            className="CategoryName"
            onChange={[Function]}
            value="Other"
          />
        </div>
      </div>
      <div
        className="Category"
      >
        <div
          className="CategoryColor"
          style={
            Object {
              "backgroundColor": "#E53F6F",
            }
          }
        />
        <div>
          <input
            className="CategoryName"
            onChange={[Function]}
            value="School"
          />
        </div>
      </div>
      <div
        className="AddCategory"
      >
        <button
          className="add-button"
          onClick={[Function]}
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
